### PR TITLE
feat: grpc for updating namespace retention period

### LINKED ISSUE #6023: Support gRPC for setting its retention period
# Rationale

This is a task of #3053 and should be done after #6022 to provide 2 gRPCs:

1. Create name space with a specified hours of data retention

```shell
influxdb_iox namespace create <namespace_name> <num_hours>
```

2. Modify retention hours of a give namespace 

```shell
influxdb_iox namespace retention <namespace_name> <num_hours>
```
'num_hours` is an integer number >=0.  We will convert this num_hours into nanoseconds and record it into the `retention_period_nano` field of the `namespace` table.

If num_hours is 0, we will record `retention_period_nano`  as NULL to say it is infinite.

This task also:
- creates a default configuration parameter INFLUXDB_IOX_RETENTION_HOURS at set it to 0 which means infinite.
- modifies the current create namespace in Router (or Ingester?) to use NFLUXDB_IOX_RETENTION_HOURS to set the value of the namespace `retention_period_nano`

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Connects to #6023

This PR support.  `influxdb_iox namespace retention --retention-hour <num_hours> <namespace>`
I will update the document after I do the second gRPC that create the namespace

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: generated_types/protos/influxdata/iox/schema/v1/service.proto [hunk 1]
```diff
@@ -5,6 +5,9 @@ option go_package = "github.com/influxdata/iox/schema/v1";
 service SchemaService {
   // Get the schema for a namespace
   rpc GetSchema(GetSchemaRequest) returns (GetSchemaResponse);
+
+  // update retention period
+  rpc UpdateNamespaceRetention(UpdateNamespaceRetentionRequest) returns (UpdateNamespaceRetentionResponse);
```
Comment (alamb) [inline generated_types/protos/influxdata/iox/schema/v1/service.proto:10]: I think this is reasonable for now. 

Longer term I recommend a  CRUD interfaces where you would pass back an entire namespace for update

That might look like 

```
  rpc GetNamespace(GetNamespaceRequest) returns (GetNamespaceResponse);
  rpc UpdateNamespace(UpdateNamespaceRequest) returns (UpdateNamespaceResponse);
...

message GetNamespaceRequest {
  // the name of the namespace
  string name = 1;
}


message UpdateNamespaceRequest {
  // the new namespace
  Namespace namespace = 1;
}
```

File: influxdb_iox/src/commands/namespace/mod.rs [hunk 2]
```diff
@@ -25,17 +27,22 @@ pub struct Config {
 enum Command {
     /// Fetch namespaces
     List,
+
+    /// Create a new namespace
```
Comment (alamb) [inline influxdb_iox/src/commands/namespace/mod.rs:31]: Maybe this should be "Update retention of an existing namespace"?

File: iox_catalog/src/interface.rs [hunk 3]
```diff
@@ -295,6 +295,13 @@ pub trait NamespaceRepo: Send + Sync {
         query_pool_id: QueryPoolId,
     ) -> Result<Namespace>;
 
+    /// Update retention period for a namespace
+    async fn update_retention_period(
+        &mut self,
+        name: &str,
+        retention_hours: i64,
```
Comment (alamb) [inline iox_catalog/src/interface.rs:302]: I find it somewhat confusing that the request contains retention in hours and the response contains retention in nanoseconds

**Review by alamb — APPROVED** [review]

File: data_types/src/lib.rs [hunk 4]
```diff
@@ -442,6 +442,8 @@ pub struct Namespace {
     /// never drop data).
     #[sqlx(default)]
     pub retention_duration: Option<String>,
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply data_types/src/lib.rs:444]: This is the unused field and string type. I plan to remove this after we use the new one

File: generated_types/protos/influxdata/iox/schema/v1/service.proto (hunk 1 above)
Comment (NGA-TRAN) [reply generated_types/protos/influxdata/iox/schema/v1/service.proto:10]: Make sense. If we need to update other fields in the future, I will make it this way

File: influxdb_iox/src/commands/namespace/mod.rs (hunk 2 above)
Comment (NGA-TRAN) [reply influxdb_iox/src/commands/namespace/mod.rs:31]: Good catch

File: iox_catalog/src/interface.rs (hunk 3 above)
Comment (NGA-TRAN) [reply iox_catalog/src/interface.rs:302]: 😄 I wanted to make it easy for user to enter in hours but for us to compute in ns. No oneobjected at the design review. If we want to change this in the future, it is very easy to do so.

**Pull request merged**